### PR TITLE
Adding `gather_ard()` function

### LINKED ISSUE #1866: Add a `get_ard()` function
This function would extract the ARDs from the gtsummary table and optionally stack them.

### PR DESCRIPTION
**Pull request by ddsjoberg**
@edelarua @Melkiades @ayogasekaram @Nolan-Steed

Can I get your opinion on the name of this function? It's a function that collects all the ARDs used to create a table. It's currently named `get_cards()`, but I think maybe there is a better name?

Here are some alternatives (but open to others as well):
- `get_ard()`: I like this because it has name symmetry for other functions like `cards::bind_ard()`, and every `ard_*()` function.
- `gather_cards()`: A part of me doesn't like the `get_*()` prefix because it reminds me of the `get()` function which gets objects from other enviornments.
- `gather_ard()`: This is just the combination of the two :)

What do you think? I think I am leaning towards `get_ard()`...

**What changes are proposed in this pull request?**
* Adding the `gather_ard()` function which collects the ARD data frames used to construct a table. (#1866)

**If there is an GitHub issue associated with this pull request, please provide link.**
closes #1866

--------------------------------------------------------------------------------

Reviewer Checklist (if item does not apply, mark is as complete)

- [ ] Ensure all package dependencies are installed: `renv::install()`
- [ ] PR branch has pulled the most recent updates from master branch: `usethis::pr_merge_main()`
- [ ] If a bug was fixed, a unit test was added.
- [ ] Run `pkgdown::build_site()`. Check the R console for errors, and review the rendered website.
- [ ] Code coverage is suitable for any new functions/features: `devtools::test_coverage()`
- [ ] `usethis::use_spell_check()` runs with no spelling errors in documentation

When the branch is ready to be merged into master:
- [ ] Update `NEWS.md` with the changes from this pull request under the heading "`# gtsummary (development version)`". If there is an issue associated with the pull request, reference it in parentheses at the end update (see `NEWS.md` for examples).
- [ ] Increment the version number using `usethis::use_version(which = "dev")` 
- [ ] Run `usethis::use_spell_check()` again
- [ ] Approve Pull Request
- [ ] Merge the PR. Please use "Squash and merge".

### CODE REVIEW

**Comment by edelarua** [comment]
A hard decision!! I think any of the 4 would work fine, but I lean slightly more toward the word `gather` rather than `get` since it is more descriptive and for the reason you mentioned. I don't think it will make much of a difference to users since they'll likely be looking up the function anyways :)

**Comment by Nolan-Steed** [comment]
I think I would vote for `gather_ard` for the same reason as it is a little more descriptive and to me it seems like a more unique name so it might be easier for users to remember.

**Comment by ddsjoberg** [comment]
`gather_ard()` it is! Thank you for the input!

File: R/gather_ard.R [hunk 1]
```diff
@@ -0,0 +1,41 @@
+#' Extract ARDs
+#'
+#' Extract the ARDs from a gtsummary table. If needed, results may be combined
+#' with `cards::bind_ard()`.
+#'
+#' @param x (`gtsummary`)\cr a gtsummary table
```
Comment (Melkiades) [inline R/gather_ard.R:6]:
```suggestion
#' @param x (`gtsummary`)\cr a gtsummary table.
```

File: R/gather_ard.R [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+#'
+#' @return list
+#' @export
+#'
+#' @examplesIf gtsummary:::is_pkg_installed('cardx', reference_pkg = 'gtsummary')
```
Comment (Melkiades) [inline R/gather_ard.R:11]: shouldnt it be present anyhow?

Comment (ddsjoberg) [reply R/gather_ard.R:11]: Nearly 100% of the time, yes. But there is a special type of R CMD Check that runs without using any of the Suggested packages, and we're meant to fail gracefully in these situations. So in this case, we just don't run the examples that require any suggested packages.

File: R/gather_ard.R [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  }
+
+  # the ARD for regression models is an additional calculation, so we don't do it by default
+  if (inherits(x, "tbl_regression")) {
+    check_pkg_installed("cardx", reference_pkg = "gtsummary")
+    return(cardx::ard_regression(x$inputs$x))
+  }
+
+  # grab ARD from standard place
+  if (!is_empty(x[["cards"]])) return(x[["cards"]]) # styler: off
```
[the remark this answers is not in the document]
Comment (ddsjoberg) [reply R/gather_ard.R:36]: Sometimes I like the single line when I find it easier to read 😉

**Review by Melkiades — APPROVED** [review]
Lgtm! Thanks Daniel ;)

**Comment by ddsjoberg** [comment]
> Lgtm! Thanks Daniel ;)
> 
> (`gather_ard` is a good name for me)

Thanks @Melkiades !

**Pull request merged**